install/zsh.go

```diff
@@ -23,9 +24,13 @@ func (z zsh) Install(cmd, bin string) error {
 
 	completeCmd := z.cmd(cmd, bin)
 	bashCompInit := "autoload -U +X bashcompinit && bashcompinit"
+	compInit := "autoload -Uz compinit && compinit"
 	if !lineInFile(z.rc, bashCompInit) {
```

Comment:
Thanks for looking Eyal.

**Let me rubber-duck for a moment:**
This line defines compInit as the string to search for / add, that same way the `bashCompInit` line is handled.
Before the suggested changes, the conditional at line 28 looks for the line in the zsh config file matching the `bashCompInit` string, then builds the `completeCmd` string to append to the file.
In my suggested changes, I added the conditional at line 31 to check for the compInit line in the zsh config and add both the `bashCompInit` and `compInit` commands plus the `completeCmd` string that may / may not have already been added to by the previous conditional.

**You are correct!** This could result in `completeCmd` string being mutated and appended like so:

```
25: completeCmd = "example_command"
[true] 29: completeCmd = "autoload -U +X bashcompinit && bashcompinit\nexample_command"
[true] 31: completeCmd = "autoload -U +X bashcompinit && bashcompinit\nexample_command\nautoload -U +X bashcompinit && bashcompinit\nexample_command"
```
I'm sorry, I will re-think this.

As for the order of the commands. I haven't tested that.
I will edit my `~/.zshrc` file and see if autocomplete works properly with the `autoload bashcompinit...` and `autoload compinit...` commands in a different order.

Kind Regards,

Ben

Reply:
I tested two versions of my `~/.zshrc` file and found that both enable autocompletion without throwing any errors.

Test A:

Test B:

Reply:
I changed line 32 from:
`completeCmd = bashCompInit + "\n" + compInit + "\n" + completeCmd`
to
`completeCmd = compInit + "\n" + completeCmd`
to avoid the issue I mentioned above.

**Test 1 - both bashcompinit and compinit lines missing from z.rc** 
(newline chars expanded for easier reading)
```
25: completeCmd = "example_command"
[True] 29: completeCmd = 
   "autoload -U +X bashcompinit && bashcompinit
   example_command"
[True] 32: completeCmd = 
   "autoload -Uz compinit && compinit
   autoload -U +X bashcompinit && bashcompinit
   example_command"
[Result] 35: completeCmd = 
   "autoload -Uz compinit && compinit
   autoload -U +X bashcompinit && bashcompinit
   example_command"
```

**Test 2 - only bashcompinit line missing from z.rc** 
(newline chars expanded for easier reading)
```
25: completeCmd = "example_command"
[True] 29: completeCmd = 
   "autoload -U +X bashcompinit && bashcompinit
   example_command"
[False] 32:  Skipped. completeCmd not modified.
[Result] 35: completeCmd = 
   "autoload -U +X bashcompinit && bashcompinit
   example_command"
```

**Test 3 - only compinit line missing from z.rc** 
(newline chars expanded for easier reading)
```
25: completeCmd = "example_command"
[False] 29: Skipped. completeCmd not modified.
[True] 32:  completeCmd = 
   "autoload -Uz compinit && compinit
   example_command"
[Result] 35: completeCmd = 
   "autoload -Uz compinit && compinit
   example_command"
```

**Test 4 - both bashcompinit and compinit lines exist in z.rc** 
(newline chars expanded for easier reading)
```
25: completeCmd = "example_command"
[False] 29: Skipped. completeCmd not modified.
[False] 32: Skipped. completeCmd not modified.
[Result] 35: completeCmd = "example_command"
```

Reply:
@posener, regarding:
> If the file already have the completeCmd line, but doesn't have the compInit in it? 
This is also an issue with the existing check for the bashCompInit line, right?

Maybe we should add a dedicated function to check if zsh has been configured and fix it before adding commands, for example:
```
func (z zsh) Configure() error {
	missingConfigLines := ""
	bashCompInit := "autoload -U +X bashcompinit && bashcompinit"
	compInit := "autoload -Uz compinit && compinit"
	if !lineInFile(z.rc, bashCompInit) {
		missingConfigLines = bashCompInit
	}
	if !lineInFile(z.rc, compInit) {
		missingConfigLines = missingConfigLines + "\n" + compInit
	}
	return appendFile(z.rc, missingConfigLines)
}
```
Sorry if above example is declared incorrectly. I'm coming from a Python background, but picking up Go while working through this change request with you. I'm not sure if I can declare `missingConfigLines` as an empty string or if returning `appendFile(z.rc, "")` would throw an exception.